src/core/network.cpp

```diff
@@ -96,6 +98,20 @@ void Network::init_indicies(){
     }
   }
 
+  //Do a cycle check and report cycles in the topology if they exist
+  std::vector<std::pair<Graph::vertex_descriptor, Graph::vertex_descriptor>> cycles;
+  detect_cycles vis(&cycles);
+  //Run an undirected, depth first search applying the loop detector visitor
+  //cycles added as vertex pairs for the closing edge of the cycle
+  boost::undirected_dfs(this->graph, vis, make_assoc_property_map(vis.vertex_coloring), make_assoc_property_map(vis.edge_coloring));
+  if(cycles.size() > 0){
+    std::string msg = "The following features form a cycle:\n";
+    for(auto p : cycles){
+      msg += "\t"+get_id(p.first)+" --> "+get_id(p.second)+"\n";
+    }
+    throw std::domain_error("Cycle(s) detected in hydrofabric\n"+msg);
```

Comment:
As phrased, this message might confusingly suggest that the identifying elements of distinct cycles are actually the set of elements in a single cycle

Reply:
Would something like
```
Cycle(s) detected in hydrofabric
The following features form the connecting edge of a cycle:
        nex-1 --> cat-0
```
be more clear?

Reply:
I think that's an improvement. Maybe "The following features each form the connecting edge of a cycle" (added 'each')?

I realize we probably don't need to word-smith this much more, since anyone encountering this is pretty likely to make their way back to us to help with getting it resolved.

Reply:
I like added "each" to be clear when multiple cycles are found.  I'll touch it up and rebase.